src/UISystem.cpp

```diff
@@ -0,0 +1,73 @@
+#include "UISystem.h"
+
+#include "AssetManager.h"
+
+UISystem::UISystem(AssetManager& asset_manager)
+    : asset_manager_(asset_manager) {
+    EventSystem::add_event_handler(EventType::LOAD_EVENT, &UISystem::handle_load, this);
+
+    // Hard-coded sample UI button mesh
+    MeshAsset::VertexData vert;
+
+    vert.position_ = { 0.0f, 0.0f, 1.0f };
+    vert.uv_ = { 0.0f, 0.0f };
+    example_mesh_.vertices_.emplace_back(vert);
```

Comment:
What's the long term plan for this?

Reply:
We would want to be loading in meshes as opposed to hardcoding them in like this. Like colour, this is gonna be phased out as well.

Reply:
I mean a more concrete plan, how is this system going to be getting these meshes in the future?

Reply:
AssetManager